### DISCUSSION TITLE
I need help with morphable relationship

### DISCUSSION THREAD

**Discussion by piotrh052b** [awcodes/filament-curator#499] [Q&A] [answered]
Hello !
I'm trying to add morphable relationship as described in documentation.

I created a migration and MediaItem model with relation:
`    public function media()
    {
        return $this->morphedByMany(MediaItem::class, 'mediable');
    }`

and in filament resource model:
`    public function media(): MorphMany
    {
        return $this->morphMany(MediaItem::class, 'mediable')
            ->orderBy('order');
    }`

When I try to save the record I get an error:

SQLSTATE[42S02]: Base table or view not found: 1146 Table 'app-alstal-dom-crm.mediables' doesn't exist

select `m052b_investments`.*, `mediables`.`media_item_id` as `pivot_media_item_id`, `mediables`.`mediable_id` as `pivot_mediable_id`, `mediables`.`mediable_type` as `pivot_mediable_type` from `m052b_investments` inner join `mediables` on `m052b_investments`.`id` = `mediables`.`mediable_id` where `mediables`.`media_item_id` in (3, 4) and `mediables`.`mediable_type` = App\Models\Investment and `m052b_investments`.`deleted_at` is null

Could someone provide me with a working solution ?

**Comment by awcodes** [comment]
there error says that it can't find the pivot table. Did you create that and run the migrations?

**Comment by piotrh052b** [comment]
Yes, as described in the documentation:

`        Schema::create('media_items', function (Blueprint $table) {
            $table->id();
            $table->morphs('mediable');
            $table->foreignId('media_id')->constrained()->onDelete('cascade');
            $table->integer('order');
            $table->string('type');
            $table->timestamps();
        });`

**Comment by awcodes** [comment, marked as answer]
Only real advice I have to check with the laravel docs to make sure your relationships are set up correctly.

**Comment by piotrh052b** [comment]
Hello !

I set the relationship in MediaItem model incorrectly - it should be "belongsTo". 
After fixing it, everything works fine.
